### DISCUSSION TITLE
serializer default value missing in nested Serializer

### DISCUSSION THREAD

**Discussion by Kurara** [encode/django-rest-framework#7825] [Potential Issue]
Version: 
Django==3.0.6
djangorestframework==3.11.0

I don't know if it's fixed in master becuase I can't upgrade all my project just to test this

## Steps to reproduce
I create a serializer with a default:
```
class MetadataFkSerializer(serializers.ModelSerializer):
    type = serializers.CharField(
        required=False, max_length=50, default='string'
    )

    class Meta:
        model = models.Metadata
        fields = (
            'type',
            'key',
            'value'
        )
```

I used it in other serializer as a list:
```
class MainSerializer(serializers.ModelSerializer):
    to_many_properties = ['metadata']
    metadata = MetadataFkSerializer(
        read_only=False,
        many=True,
    )
    class Meta:
        model = models.Main
        fields = (
            [...]
            'metadata'
        )
```
## Expected behavior
In the save of the MainSerilizer, I'm custom saving the metadata as it's a list. When I'm in POST I find the `_metadata0.get('type')` filled with 'string' but in PATCH no.
```
 _metadata = self.to_many_objects.get('metadata')
 if _metadata is not None:
     for _metadata0 in _metadata:
         models.Metadata.objects.create(
             type=_metadata0.get('type'), # Here the default must be returned if nothing expecified in http body
             key=_metadata0.get('key'),
             value=_metadata0.get('value'),
             main_model=model
         )
```

For you information, this to_many_objects is filled here:
```
def update(self, instance, validated_data):
    # 1. Extract to_many data
    _model_data = validated_data.copy()
    for _to_many in self.to_many_properties:
        if _to_many in _model_data:
            _model_obj = _model_data.pop(_to_many)
            self.to_many_objects[_to_many] = _model_obj
```
So it comes from `validated_data` Seems in POST validated_data has the default, but in PATCH no.
## Actual behavior
`_metadata0.get('type')` is not filled with 'string' in PATCH

I fixed it in my project just by doing: `_metadata0.get('type', 'string')` but I think this should be reported as I don't think is a correct way to behave for the framework.

**Comment by lovelydinosaur** [comment]
This looks a bit too convoluted for me to be able to dig into / help with.

**Reply by Kurara** [reply]
Sorry I only realized about the answer now (2 years later). How Can I help you? I could create a repository with this issue where you can test it. Maybe I create it and I realize it's not happening anymore
